traits/has_traits.py

```diff
@@ -2562,8 +2562,15 @@ def on_trait_change ( self, handler, name = None, remove = False,
 
         if isinstance( name, list ):
             for name_i in name:
-                self.on_trait_change( handler, name_i, remove, dispatch,
-                                      priority, target )
+                self.on_trait_change(
+                    handler,
+                    name=name_i,
+                    remove=remove,
+                    dispatch=dispatch,
+                    priority=priority,
+                    deferred=deferred,
```

Comment:
I think this is not meant to be here?

Reply:
You're right - wasn't aware of the deferred option and it isn't present in the call on line 2555 above and appears not to be documented anywhere.  In fact deferred seems to do nothing. Is this a dead part of the API?

Reply:
The line 2555 call is to a different method: `_on_trait_change`, not `on_trait_change`.

Don't know what `deferred` is *supposed* to do; I can open a separate issue for that.

Reply:
Looks like `deferred` is tied into the `post_init` machinery. I'd be nervous about messing with it in this PR.

Reply:
Hmmm... it is used below.  Apologies for the noise.  Clearly I need to work out up what it actually does.